plugin-scalebar/src/main/java/com/mapbox/maps/plugin/scalebar/ScaleBarImpl.kt

```diff
@@ -312,20 +301,53 @@ class ScaleBarImpl : ScaleBar, View {
     canvas.drawText(text, x, y, textPaint)
   }
 
+  /**
+   * @return A tuple with the scale bar segment distance and the amount of bar segments
+   */
+  @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+  internal fun findSuitableScaleBarSegments(maxDistance: Float): Pair<Float, Int> {
+    var pair: Pair<Int, Int> = scaleTable[0]
+    for (i in 1 until scaleTable.size) {
+      pair = scaleTable[i]
+      val distance = pair.first
+      if (distance > maxDistance) {
+        // use the last scale here, otherwise the scale will be too large
+        pair = scaleTable[i - 1]
+        break
+      }
+    }
+    val distance = pair.first.toFloat()
+    var rectCount = pair.second
+    var unitDistance = distance / rectCount
+    // When maxDistance is small (i.e. high zoom levels near the poles) then
+    // the `distance` might be bigger than maxDistance. This loop will keep removing
+    // bar divisions (rectCount) until it fits
+    while (unitDistance * rectCount > maxDistance && rectCount > 0) {
+      rectCount--
+    }
+    // In case the unitDistance doesn't fit at all we fallback to maxDistance (rounded to 1
+    // decimal) with 1 division
+    if (rectCount == 0) {
+      unitDistance = (maxDistance * 10).toInt() / 10.0F
+      rectCount = 1
+    }
+    return Pair(unitDistance, rectCount)
+  }
+
   /**
    * Get the formatted distance text according unit and distance
    *
    * @param distance original distance
    * @return Formatted distance text
    */
-  internal fun getDistanceText(distance: Int): String {
-    return if (distance == 0) {
+  internal fun getDistanceText(distance: Float): String {
+    return if (distance == 0F) {
       "0"
     } else if (METER_UNIT == unit) {
```

Comment:
nit:
```suggestion
    } else {
       val (unitMinDistance, unitName) = if (METE_UNIT == unit) { KILOMETER to KILOMETER_UNIT } else { FEET_PER_MILE to MILE_UNIT }
        return if (distance < unitMinDistance) {
           decimalFormat.format(distance) + unit
        } else {
           decimalFormat.format(distance * 1.0 / unitMinDistance) + unitName
       }
      } 
```

Reply:
Somehow I see your suggestion as harder/more difficult to read. (nit: moreover, it will create a new object (`Pair`) every time is called)

Maybe others can also give their opinion.

Reply:
I'm fine with the current notation